src/main.c

```diff
@@ -224,6 +242,35 @@ poe_cmd_send(struct cmd *cmd)
 	return 0;
 }
 
+static int
+poe_i2c_cmd_send(struct cmd *cmd, unsigned char *reply, int len)
```

Comment:
Seems like we should separate both the serial and I2C implementations out, so that main.c can be purely about sending frames, not frames *through a specific medium*.

Reply:
Yes, we should have a backend. It would also solve the issue of the synchronous/asynchronous behaviour. If the backend would be in a separate thread, we could have everything asynchronous.